src/java_bytecode/java_types.h

```diff
@@ -579,6 +579,13 @@ class java_generic_symbol_typet : public symbol_typet
 public:
   typedef std::vector<reference_typet> generic_typest;
 
+  /// Construct a generic symbol type from base_ref and class_name_prefix which
+  /// must correspond to a generic type
+  /// (see \ref java_generic_symbol_typet::java_type_from_string).
+  /// For instance `java_generic_symbol_typet(symbol_typet("Map"),
+  /// "Ljava/util/Map<TK;TV;>;", "java/util/Map")` will generate a symbol type
+  /// with identifier "Map", and two generic types with identifier
+  /// "java/util/Map::K" and "java/util/Map::V" respectively.
```

Comment:
To reflect what this does a bit better I would say - Construct a generic symbol type by extending the symbol type \p type with generic types extracted from the reference \p base_ref. This assumes that the class with name \p class_name_prefix extends/implements the class in \p type.

For the example you give, I would change the third input to "java/util/HashMap" and then the generic types identifiers become "java.util.HashMap::K" and "java.util.HashMap::V".

Reply:
Actually, there should be a precondition in the constructor that checks that the identifier of `type` and the identifier of the type created for `base_ref` are the same. But that is up to you if you want to add this here, I can do that in a separate PR.